compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -2537,4 +2564,166 @@ public int hashCode() {
     private interface TypeEqualityPredicate {
         boolean test(BType source, BType target, Set<TypePair> unresolvedTypes);
     }
+
+    public boolean hasFillerValue(BType type) {
+        if (type.tag < TypeTags.RECORD) {
+            return true;
+        }
+        switch (type.tag) {
+            case TypeTags.MAP:
+            case TypeTags.ANY:
+                return true;
+            case TypeTags.ARRAY:
+                return checkFillerValue((BArrayType) type);
+            case TypeTags.FINITE:
+                return checkFillerValue((BFiniteType) type);
+            case TypeTags.UNION:
+                return checkFillerValue((BUnionType) type);
+            case TypeTags.OBJECT:
+                return checkFillerValue((BObjectType) type);
+            case TypeTags.RECORD:
+                return checkFillerValue((BRecordType) type);
+            case TypeTags.TUPLE:
+                BTupleType tupleType = (BTupleType) type;
+                return tupleType.getTupleTypes().stream().allMatch(eleType -> hasFillerValue(eleType));
+            default:
+                return false;
+        }
+    }
+
+    private boolean checkFillerValue(BObjectType type) {
+        BAttachedFunction initFunction = ((BObjectTypeSymbol) type.tsymbol).initializerFunc;
+        if (initFunction == null) {
+            if ((type.tsymbol.flags & Flags.ABSTRACT) == Flags.ABSTRACT) {
+                return false;
+            }
```

Comment:
This can be

```java
    return (type.tsymbol.flags & Flags.ABSTRACT) != Flags.ABSTRACT;
```

right?

Reply:
Even if the object is not an abstract object we still need to do some checks or parameters in the initializer function. But I changed the logic giving importance to abstraction check. Can you check the new logic whether I got your point correctly? https://github.com/ballerina-platform/ballerina-lang/pull/20749/commits/ff0acc05501340c2064c3bb1b3211752955782ad